lib/sqlalchemy/dialects/mssql/base.py

```diff
@@ -2344,8 +2344,20 @@ def get_column_specification(self, column, **kwargs):
         colspec = self.preparer.format_column(column)
 
         # type is not accepted in a computed column
-        if column.computed is not None:
+        if (
+            column.computed is not None
+            and not column.computed._system_versioning
+        ):
             colspec += " " + self.process(column.computed)
+        elif column.computed is not None:
+            colspec += (
+                " "
+                + self.dialect.type_compiler_instance.process(
```

Comment:
**Federico Caselli** (CaselIT) wrote:

mssql is really picky here!

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/3803